library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java

```diff
@@ -469,9 +488,10 @@ private void flushCodec() throws ExoPlaybackException {
     decodeOnlyPresentationTimestamps.clear();
     // Workaround for framework bugs.
     // See [Internal: b/8347958], [Internal: b/8578467], [Internal: b/8543366].
-    if (Util.SDK_INT >= 18) {
+    if (Util.SDK_INT >= 18 && codecReinitState == REINIT_STATE_NONE) {
```

Comment:
Taking a final pass at this, I'm not sure I follow the logic here. Shouldn't this be left as it was, and then codecReinitState should be set to REINIT_STATE_NONE at the bottom of this method, where you set hasQueuedOneInputBuffer to false?